README.md

```diff
@@ -1,3 +1,105 @@
+
+
+## 2. Set up
+
+Before you can start using the SDK, you will need to set up the SDK keys. Most of these keys can be found in your account's [dashboard](https://console.visenze.com/).
+
+First, take a look at the table below to understand what each key represents:
+
+| Key | Importance | Description |
+|:---|:---|:---|
+| app_key | Compulsory | All SDK functions depends on a valid app_key being set. The app key also limits the API features you can use. |
+| placement_id | Compulsory | Your placement id. |
+| timeout | Optional | Timeout for APIs in ms. Defaulted to 15000 |
+
+To create a ProductSearch instance:
+
+```dart
+const psClient = await VisenzeProductSearch.create('APP_KEY', 'PLACEMENT_ID');
+```
+
+## 3. APIs
+
+### 3.1 Search by image
+POST /product/search_by_image
+
+Searching by Image can happen in three different ways - by url, id or File.
+
+- Using image id:
+
+  ```dart
+  var params = {
+    im_id: 'your-image-id'
+  };
+
+  var response = await psClient.productSearchByImage(params);
+  ```
+
+- Using image url:
+
+  ```dart
+  var params = {
+    im_url: 'your-image-url'
+  };
+
+  var response = await psClient.productSearchByImage(params);
+  ```
+
+- Using image file:
+
+  ```dart
+  var body = {
+    image: imageObjFile
+  };
+
+  var response = await psClient.productSearchByImage(params);
+  ```
+
+> The request parameters for this API can be found at [ViSenze Documentation Hub](https://ref-docs.visenze.com/reference/search-by-image-api-1).
+
+### 3.2 Recommendationss
+GET /product/recommendations/{product_id}
+
+Search for visually similar products in the product database giving an indexed product’s unique identifier.
+
+```dart
+var productId = 'your-product-id';
+
+// example paramters
+var parameters = {
+  limit: 20 // limit results returned to 20 results
+};
+
+var response = await psClient.productSearchById(productId, parameters);
+```
+> The request parameters for this API can be found at [ViSenze Documentation Hub](https://ref-docs.visenze.com/reference/visually-similar-api).
+
+## 4. Search results
+
+This sdk use the [http](https://pub.dev/packages/http) library for making API request. The response is a http [Response](https://pub.dev/documentation/http/latest/http/Response-class.html).
+
+The list of error codes can be found [here](https://ref-docs.visenze.com/reference/error-codes).
+For response properties, please refer to the APIs docs.
+
+```dart
+var productId = 'your-product-id';
+var response = await psClient.productSearchById(productId);
+
+if (response.statusCode == 200) {
+  Map<String, dynamic> successResp = jsonDecode(response.body);
+  List<Map<String, dynamic>> results = successResp['result'];
+  // handle results here
+} else {
+  // handle error here
+}
+```
```

Comment:
instructions to send events / send batch event / get uid / get sid / set sid / get last query id

Reply:
instructions on image resizing

Reply:
instructions on how to install / build flutter sdk into developer application